src/sharepoint/rest/Queryable.ts

```diff
@@ -220,6 +220,22 @@ export class QueryableCollection extends Queryable {
         this._query.add("$select", selects.join(","));
         return this;
     }
+    public expand(...expands: string[]): QueryableCollection {
+        this._query.add("$expand", expands.join(","));
+        return this;
+    }
+    public orderby(...orderbys: string[]): QueryableCollection {
+        this._query.add("$orderby", orderbys.join(","));
+        return this;
+    }
+    public skip(skip: number): QueryableCollection {
```

Comment:
hmm, I didn't even know we had an items collection!
list.getitemsbycammlquery is currently returning a Promise<any>:
should that be a Promise<Items>:

sorry for the repeated ?s . just trying to understand the library.

I'll look to move $skip and $top to the Items collection(although it was
working in queryable)




On Mon, May 9, 2016 at 7:42 PM, Patrick Rodgers <notifications@github.com>
wrote:

> In src/sharepoint/rest/Queryable.ts
> <https://github.com/OfficeDev/PnP-JS-Core/pull/97#discussion_r62591983>:
>
> > @@ -220,6 +220,22 @@ export class QueryableCollection extends Queryable {
> >          this._query.add("$select", selects.join(","));
> >          return this;
> >      }
> > +    public expand(...expands: string[]): QueryableCollection {
> > +        this._query.add("$expand", expands.join(","));
> > +        return this;
> > +    }
> > +    public orderby(...orderbys: string[]): QueryableCollection {
> > +        this._query.add("$orderby", orderbys.join(","));
> > +        return this;
> > +    }
> > +    public skip(skip: number): QueryableCollection {
>
> For Skip I did see this
> <https://msdn.microsoft.com/en-us/library/office/dn531433.aspx#sectionSection1>
> :
>
> The OData $skip query option does not work when querying items of a
> particular list. It does work when querying the lists endpoint.
>
> So I am not sure adding skip to the queryable collection is correct - OR
> we need to do some overriding in the items collection. Not sure of the best
> way to handle this.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/OfficeDev/PnP-JS-Core/pull/97/files/47bc9ecd6098b86943e63c44412eb8b446c0c16c#r62591983>
>